java/Oleksii-Riabov/src/com/zoolatech/lecture4/tasks/_2/Tree.java

```diff
@@ -0,0 +1,50 @@
+package com.zoolatech.lecture4.tasks._2;
+
+public class Tree {
+
+    class Node {
+    int value;
+    Node leftNode;
+    Node rightNode;
+
+    Node(int value) {
+        this.value = value;
+        rightNode = null;
+        leftNode = null;
+    }
+}
+
+    Node root;
+
+    private Node addRecursive(Node current, int value) {
+        if (current == null) {
+            return new Node(value);
+        }
+
+        if (value < current.value) {
+            current.leftNode = addRecursive(current.leftNode, value);
+        } else if (value > current.value) {
+            current.rightNode = addRecursive(current.rightNode, value);
+        } else {
```

Comment:
I think you can either remove the else block or remove the last return statement since they duplicate each other

Reply:
"Else' bloc has been removed.